blockchain/state/database.go

```diff
@@ -45,11 +45,11 @@ const (
 type Database interface {
 	// OpenTrie opens the main account trie.
 	OpenTrie(root common.Hash) (Trie, error)
-	OpenTrieForPrefetching(root common.Hash) (Trie, error)
+	OpenTrieWithOpts(root common.Hash, opts *statedb.TrieOpts) (Trie, error)
```

Comment:
How about merging these methods?

Reply:
We could, but still New-NewWithOpts, NewTrie-NewTrieWithOpts exists. I'd like to keep OpenTrie-OpenTrieWithOpts as well for consistency.

Reply:
Yes, I meant merging all of the `-WithOpts` methods. `NewTrie-NewTrieWithOpts`, `OpenTrie-OpenTrieWithOpts `, and `OpenStorageTrie-OpenStorageTrieWithOpts`. Passing parameter `opts` would be enough for the readers to understand. It looks clearer to me. What do you think?

Reply:
I agree that having only one interface is clearer. Let me push a commit that merges the two and see if it's okay.

Reply:
@jeongkyun-oh Added commit 7a7081f89c80e6dde1f2013d5d7673f458fe84e9. I my perspective, too many files are affected..

Reply:
Hmm.. I still think that merging is better, but it would be better to create separate PR maybe? Thanks for updating though.

Reply:
I am good with them merged. Thank you for the guidance.